src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java

```diff
@@ -74,13 +78,33 @@ private void changeDueDate(RoutingContext routingContext) {
       .after(r -> getExistingLoan(loanRepository, r))
       .thenApply(LoanValidator::refuseWhenLoanIsClosed)
       .thenApply(this::toLoanAndRelatedRecords)
+      .thenComposeAsync(r -> r.after(requestQueueRepository::get))
+      .thenApply(r -> r.map(this::unsetDateTruncationFlagIfNoOpenRecallsInQueue))
       .thenApply(itemStatusValidator::refuseWhenItemStatusDoesNotAllowDueDateChange)
       .thenApply(r -> changeDueDate(r, request))
       .thenComposeAsync(r -> r.after(loanRepository::updateLoan))
       .thenComposeAsync(r -> r.after(eventPublisher::publishDueDateChangedEvent))
       .thenApply(r -> r.next(scheduledNoticeService::rescheduleDueDateNotices))
       .thenCompose(r -> r.after(loanNoticeSender::sendManualDueDateChangeNotice));
   }
+  
+  private LoanAndRelatedRecords unsetDateTruncationFlagIfNoOpenRecallsInQueue(
+      LoanAndRelatedRecords loanAndRelatedRecords) {
+
+    RequestQueue queue = loanAndRelatedRecords.getRequestQueue();
+    Loan loan = loanAndRelatedRecords.getLoan();
+    if (loan.wasDueDateChangedByRecall() && !hasOpenRecalls(queue)) {
+      return loanAndRelatedRecords.withLoan(loan.unsetDueDateChangedByRecall());
+    }
+    else {
+      return loanAndRelatedRecords;
+    }
+  }
+
+  private Boolean hasOpenRecalls(RequestQueue queue) {
+    return queue.getRequests().stream()
+        .anyMatch(request -> request.getRequestType() == RequestType.RECALL && request.isNotYetFilled());
+  }
```

Comment:
I suggest moving this to the RequestQueue. It helps to hide the details of the request queue and aid reuse across this and the renewal.

Reply:
good idea, will do.